numpyro/distributions/continuous.py

```diff
@@ -462,52 +465,156 @@ def log_prob(self, value):
 
 
 @copy_docs_from(Distribution)
-class Normal(Distribution):
+class LogNormal(TransformedDistribution):
     arg_constraints = {'loc': constraints.real, 'scale': constraints.positive}
-    support = constraints.real
     reparametrized_params = ['loc', 'scale']
 
     def __init__(self, loc=0., scale=1., validate_args=None):
-        self.loc, self.scale = promote_shapes(loc, scale)
-        batch_shape = lax.broadcast_shapes(np.shape(loc), np.shape(scale))
-        super(Normal, self).__init__(batch_shape=batch_shape, validate_args=validate_args)
+        base_dist = Normal(loc, scale)
+        self.loc, self.scale = base_dist.loc, base_dist.scale
+        super(LogNormal, self).__init__(base_dist, ExpTransform(), validate_args=validate_args)
+
+    @property
+    def mean(self):
+        return np.exp(self.loc + self.scale ** 2 / 2)
+
+    @property
+    def variance(self):
+        return (np.exp(self.scale ** 2) - 1) * np.exp(2 * self.loc + self.scale ** 2)
+
+
+def _batch_mahalanobis(bL, bx):
+    # NB: The following procedure handles the case: bL.shape = (i, 1, n, n), bx.shape = (i, j, n)
+    # because we don't want to broadcast bL to the shape (i, j, n).
+
+    # Assume that bL.shape = (i, 1, n, n), bx.shape = (..., i, j, n),
+    # we are going to make bx have shape (..., 1, j,  i, 1, n) to apply batched tril_solve
+    sample_ndim = bx.ndim - bL.ndim + 1  # size of sample_shape
+    out_shape = np.shape(bx)[:-1]  # shape of output
+    # Reshape bx with the shape (..., 1, i, j, 1, n)
+    bx_new_shape = out_shape[:sample_ndim]
+    for (sL, sx) in zip(bL.shape[:-2], out_shape[sample_ndim:]):
+        bx_new_shape += (sx // sL, sL)
+    bx_new_shape += (-1,)
+    bx = np.reshape(bx, bx_new_shape)
+    # Permute bx to make it have shape (..., 1, j, i, 1, n)
+    permute_dims = (tuple(range(sample_ndim))
+                    + tuple(range(sample_ndim, bx.ndim - 1, 2))
+                    + tuple(range(sample_ndim + 1, bx.ndim - 1, 2))
+                    + (bx.ndim - 1,))
+    bx = np.transpose(bx, permute_dims)
+
+    # reshape to (-1, i, 1, n)
+    xt = np.reshape(bx, (-1,) + bL.shape[:-1])
+    # permute to (i, 1, n, -1)
+    xt = np.moveaxis(xt, 0, -1)
+    solve_bL_bx = solve_triangular(bL, xt, lower=True)  # shape: (i, 1, n, -1)
+    M = np.sum(solve_bL_bx ** 2, axis=-2)  # shape: (i, 1, -1)
+    # permute back to (-1, i, 1)
+    M = np.moveaxis(M, -1, 0)
+    # reshape back to (..., 1, j, i, 1)
+    M = np.reshape(M, bx.shape[:-1])
+    # permute back to (..., 1, i, j, 1)
+    permute_inv_dims = tuple(range(sample_ndim))
+    for i in range(bL.ndim - 2):
+        permute_inv_dims += (sample_ndim + i, len(out_shape) + i)
+    M = np.transpose(M, permute_inv_dims)
+    return np.reshape(M, out_shape)
+
+
+@copy_docs_from(Distribution)
+class MultivariateNormal(Distribution):
+    arg_constraints = {'loc': constraints.real,
+                       'covariance_matrix': constraints.positive_definite,
```

Comment:
Since this is adding lazy parameters to `arg_constraints` we should also make a change in `Distribution.__init__` to only check parameters that are there in the instances `__dict__` 

```python
if param not in self.__dict__ and isinstance(getattr(type(self), param), lazy_property):
    continue 
```

Reply:
Thanks for providing an explicit code! It is very helpful! :D